htdocs/core/lib/project.lib.php

```diff
@@ -354,13 +354,17 @@ function project_admin_prepare_head()
  * @param   int         $projectidfortotallink  0 or Id of project to use on total line (link to see all time consumed for project)
  * @param   string      $filterprogresscalc     filter text
  * @param   string      $showbilltime           Add the column 'TimeToBill' and 'TimeBilled'
+ * @param   array       $arrayfields            Array with displayed coloumn information
  * @return	int									Nb of tasks shown
  */
-function projectLinesa(&$inc, $parent, &$lines, &$level, $var, $showproject, &$taskrole, $projectsListId = '', $addordertick = 0, $projectidfortotallink = 0, $filterprogresscalc = '', $showbilltime = 0)
+function projectLinesa(&$inc, $parent, &$lines, &$level, $var, $showproject, &$taskrole, $projectsListId = '', $addordertick = 0, $projectidfortotallink = 0, $filterprogresscalc = '', $showbilltime = 0, $arrayfields = array())
 {
-	global $user, $langs, $conf, $db;
+	global $user, $langs, $conf, $db, $hookmanager;
 	global $projectstatic, $taskstatic;
 
+	$extrafields = new ExtraFields($db);
```

Comment:
The method projectLinesa can be called a lot of time (at list for every line).
For this reason we should not reload the definition of extrafields here (this need useless database access). With v11, the $extrafields is a global variable so you can make the init it once with 
$extrafields->fetch_name_optionals_label($taskstatic->table_element);
in the page that call the projectLinesa and got it in method 
global $extrafields

Reply:
@eldy ok, done